src/circuit/gadgets/constraints.rs

```diff
@@ -1,6 +1,8 @@
 // Initially taken from: rust-fil-proofs/storage-proofs-core/src/gadgets/
 
+use super::pointer::AllocatedPtr;
```

Comment:
`constraints.rs` shouldn't use `AllocatedPtr`. More generally, it shouldn't contain anything that depends on the Lurk data model.